integrations/terraform-cloud/client.py

```diff
@@ -97,9 +97,22 @@ async def get_paginated_resources(
     async def get_paginated_organizations(
         self,
     ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        cache_key = "terraform-cloud-organization"
+        if cache := event.attributes.get(cache_key):
+            logger.info("Retrieving organizations data from cache")
+            yield cache
+            return
+
+        all_organizations = []
         logger.info("Fetching organizations")
         async for organizations in self.get_paginated_resources("organizations"):
             yield organizations
+            all_organizations.extend(organizations)
```

Comment:
first append it and the yield back
```suggestion
            all_organizations.extend(organizations)
            yield organizations
```